hs_access_control/models/resource.py

```diff
@@ -79,30 +122,59 @@ def edit_users(self):
 
         This is a property so that it is a workalike for a prior explicit list.
 
+        This now accounts for group and community privileges
+
         If the resource is immutable, an empty QuerySet is returned.
+
         """
 
         if self.immutable:
             return User.objects.none()
         else:
             return User.objects\
-                       .filter(Q(is_active=True) &
-                               (Q(u2urp__resource=self.resource,
-                                  u2urp__privilege__lte=PrivilegeCodes.CHANGE) |
-                                Q(u2ugp__group__gaccess__active=True,
-                                  u2ugp__group__g2grp__resource=self.resource,
-                                  u2ugp__group__g2grp__privilege__lte=PrivilegeCodes.CHANGE)))\
-                .distinct()
+                       .filter(self.__edit_users_from_individual |
+                               self.__edit_users_from_group |
+                               self.__edit_users_from_community).distinct()
+
+    @property
+    def __view_groups_from_group(self):
+        return Q(gaccess__active=True,
+                 g2grp__resource=self.resource)
+
+    @property
+    def __view_groups_from_community(self):
+        return Q(gaccess__active=True,
+                 g2gcp__community__c2gcp__allow_view=True,
```

Comment:
`g2gcp__allow_view=True`?

Reply:
The flag allow_view has to apply to the group the resource can view. The resource can be viewed by g2gcp__community__c2gcp__group. Not g2gcp__group.